### DISCUSSION TITLE
any solution for graphql subscriptions?

### DISCUSSION THREAD

**Discussion by Nisthar** [directus/directus#15202] [Archive]
Since directus support graphql but not graphql subscriptions. Are there any solution to add graphql subscriptions? is it worth using supabase for just graphql subscriptions?

**Comment by azrikahar** [comment]
> Are there any solution to add graphql subscriptions?

It is planned to be supported in the upcoming WebSockets feature. 👍

> is it worth using supabase for just graphql subscriptions?

As all things in dev, I think often times the answer is "it depends!" 😄 Note that this response is not specifically regarding Supabase, but just solutions in general. 

There are always tradeoffs whenever we use multiple solutions within an architecture to solve business requirements or use cases, and the perceived "worthiness" often times can depend on other variables such as budget, resources, time, maintainability etc. For example if there is an immediate need to having graphql subscriptions for your application, then you would indeed need to use a different solution to fill in for that use case in the meantime. There is also a possibility of "migrating" graphql subscriptions back to Directus later on for purposes such as simplifying the overall architecture or just one less moving part to consider, but then that'd require time and resources to do so.

**Reply by Nisthar** [reply]
is the websocket feature already planned or not? is there a eta on it?

**Reply by azrikahar** [reply]
> is the websocket feature already planned or not?

Yes it is, ref: #14737

>  is there a eta on it?

I don't have an actual ETA at hand personally, but development is underway (as seen in that PR) and I reckon that's more tangible than `#soon` 👍
